app/controllers/configuration_controller.rb

```diff
@@ -162,10 +162,20 @@ def timeprofile_set_days_hours(_timeprofile = @timeprofile)
     @timeprofile_details = {}
     @timeprofiles.each do |timeprofile|
       @timeprofile_details[timeprofile.description] = {}
-      @timeprofile_details[timeprofile.description][:days] =
+      if timeprofile.profile[:days].nil?
+        @timeprofile_details[timeprofile.description][:days] =
+        timeprofile.profile["days"].collect { |day| DateTime::ABBR_DAYNAMES[day.to_i] }
+      else
+        @timeprofile_details[timeprofile.description][:days] =
         timeprofile.profile[:days].collect { |day| DateTime::ABBR_DAYNAMES[day.to_i] }
```

Comment:
Since these 2 branches are basically the same except for the one key, it's better to keep it DRY and only extract the part that is different - something like the following is preferable

```suggestion
      profile_key = timeprofile.profile[:days].nil? ? "days" : :days
      @timeprofile_details[timeprofile.description][profile_key] =
        timeprofile.profile["days"].collect { |day| DateTime::ABBR_DAYNAMES[day.to_i] }
```